primitiv/_function.pyx

```diff
@@ -17,6 +17,10 @@ class functions:
 
     @staticmethod
     def raw_input(shape, vector[float] data, Device device = None, Graph graph = None):
+        if device is None:
```

Comment:
`get_cpp_something` may return NULL and NULL is acceptable by C++ APIs (if they received NULLs, it is replaced to the C++-side default objects). This may confuse the overall behavior.
It is good to add some Python-side functions `get_default_or_die`, or adding an option to `get_default` to raise an exception if the Python-side default object is null.

Reply:
Fixed. `get_default()` checks default object now.